EpsilonLights/Tests/tests/LightsTest/LightsTest.c

```diff
@@ -0,0 +1,391 @@
+
+    TEST_ASSERT_EQUAL_MESSAGE(0, prevSigState, "prevSigState is supposed to be 0");
+}
+
+void test_blinkSignalLights_sigLightsLeftONSigLightsRightONPrevSigStateZero()
+{
+    uint32_t prevWakeTime = 0;
+    uint8_t prevSigState = 0;
+    uint32_t blinkerTimer = 0;
+
+    sigLightsHandle.left = 1;
+    sigLightsHandle.right = 1;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, sigLightsHandle.right);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, sigLightsHandle.left);
+
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+    TEST_ASSERT_EQUAL_MESSAGE(0, blinkerTimer, "blinkerTimer is supposed to be 0");
+    TEST_ASSERT_EQUAL_MESSAGE(1, prevSigState, "prevSigState is supposed to be 1");
+}
+
+void test_blinkSignalLights_prevSigState1SigLightsLeftONSigLightsRightON()
+{
+    uint32_t prevWakeTime = 0;
+    uint8_t prevSigState = 1;
+
+    sigLightsHandle.left = 2;
+    sigLightsHandle.right = 2;
+
+    keepBlinkersOFF_resetBlinkerTimerToZero(prevWakeTime, prevSigState);
+    turnBlinkersON_updateTimer(prevWakeTime, prevSigState);
+    keepBlinkersOFF_updateTimer(prevWakeTime, prevSigState);
+}
+
+void keepBlinkersOFF_resetBlinkerTimerToZero(uint32_t prevWakeTime, uint8_t prevSigState)
+{
+    uint32_t blinkerTimer = 771;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, LIGHT_OFF);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, LIGHT_OFF);
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+    TEST_ASSERT_EQUAL_MESSAGE(0, blinkerTimer, "blinkerTimer is supposed to be 0");
+}
+
+void turnBlinkersON_updateTimer(uint32_t prevWakeTime, uint8_t prevSigState)
+{
+    uint32_t blinkerTimer = 300;
+    uint32_t expectedBlinkerTimer = blinkerTimer + 10;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, sigLightsHandle.right);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, sigLightsHandle.left);
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+    TEST_ASSERT_EQUAL_MESSAGE(expectedBlinkerTimer, blinkerTimer, "blinkerTimer is supposed to be expectedBlinkerTimer");
+}
+
+void keepBlinkersOFF_updateTimer(uint32_t prevWakeTime, uint8_t prevSigState)
+{
+    uint32_t blinkerTimer = 400;
+    uint32_t expectedBlinkerTimer = blinkerTimer + 10;
+
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, LIGHTS_UPDATE_FREQ, 0);
+    HAL_GPIO_WritePin_Expect(RSIGNAL_GPIO_Port, RSIGNAL_Pin, LIGHT_OFF);
+    HAL_GPIO_WritePin_Expect(LSIGNAL_GPIO_Port, LSIGNAL_Pin, LIGHT_OFF);
+    blinkSignalLights(&prevWakeTime, &blinkerTimer, &prevSigState);
+    TEST_ASSERT_EQUAL_MESSAGE(expectedBlinkerTimer, blinkerTimer, "blinkerTimer is supposed to be expectedBlinkerTimer");
+}
+
+// TESTS for updateStrobeLight1
```

Comment:
Please write some comment for all your test cases. Explain the situation you're testing and what is the expected outcome. Try to avoid straight up saying the value of the variable if you can.